src/clj/rems/db/user_secrets.clj

```diff
@@ -31,9 +31,9 @@
 (defn update-user-secrets! [user new-secrets]
   (assert user "User missing!")
   (let [old-secrets (get-user-secrets user)
-        validated (try (validate-user-secrets new-secrets) (catch Exception _e nil))]
-    (if (= (set (keys validated))
-           (set (keys new-secrets)))
+        validated (try (validate-user-secrets (merge old-secrets new-secrets))
+                       (catch Exception _e nil))]
```

Comment:
should probably have some sort of `{:success false}` logic here (just like in user-settings) instead of failing silently.

Reply:
The next line is the `if` that goes to `{:success false}` if we got the `nil` here?

Reply:
Oh right, I'm blind.